src/librustc_codegen_llvm/attributes.rs

```diff
@@ -77,9 +77,37 @@ pub fn set_instrument_function(cx: &CodegenCx<'ll, '_>, llfn: &'ll Value) {
     if cx.sess().instrument_mcount() {
         // Similar to `clang -pg` behavior. Handled by the
         // `post-inline-ee-instrument` LLVM pass.
+
+        // The function name varies on platforms.
+        // See test/CodeGen/mcount.c in clang.
+        let mcount_name = if cfg!(target_os = "netbsd") {
```

Comment:
`cfg` is wrong here, because it will be checking for the host (the computer running the compiler) rather than for the target (whatever is intended to run the final executable).

You should be checking for stuff in `sess().target` or, even better, make this name a part of target definitions and use that value directly.

Reply:
> make this name a part of target definitions and use that value directly.

The targets are defined in `src/librustc_target/spec/`(e.g. like this)? If so, I should add `target_mcount` or something in `Ok(Target {..})`, right?

https://github.com/rust-lang/rust/blob/master/src/librustc_target/spec/aarch64_unknown_netbsd.rs

Reply:
> If so, I should add target_mcount or something in Ok(Target {..}), right?

Yeah.